make fails on ubuntu 14.04 in Vagrant

While running `make` on the default ubuntu 14.04 vagrant box with all dependencies installed:

```
[snip]
> link: capnp/endian-test.o
> test: capnp/endian-test
> link: capnp/endian-reverse-test.o
> test: capnp/endian-reverse-test
> link: capnp/endian-fallback-test.o
> test: capnp/endian-fallback-test
> install: node-capnp/node-capnp.ekam-manifest
> terminate called after throwing an instance of 'std::logic_error'
>   what():  Already fulfilled this promise.
> Aborted (core dumped)
> make: *** [tmp/.ekam-run] Error 134
```

using Ubuntu clang version 3.5.0-4ubuntu2~trusty1 (tags/RELEASE_350/final) (based on LLVM 3.5.0)


using

I currently consistently get this error with NFS and 14.04 and clang-3.5 (as well as clang-3.4):

@kentonv does this ring any bells to you? Going to see if I get it in my Debian system outside VirtualBox now.

```
compile: sandstorm/union-fs.c++
  In file included from /ekam-provider/canonical/sandstorm/union-fs.c++:17:
  /ekam-provider/canonical/sandstorm/union-fs.h:23:10: fatal error: 'sandstorm/package.capnp.h' file not found
  #include <sandstorm/package.capnp.h>
           ^
  1 error generated.
compile: sandstorm/sandstorm-http-bridge.c++
  /ekam-provider/canonical/sandstorm/sandstorm-http-bridge.c++:45:10: fatal error: 'sandstorm/grain.capnp.h' file not found
  #include <sandstorm/grain.capnp.h>
           ^
  1 error generated.
capnpc: sandstorm/supervisor.capnp
  sandstorm/grain.capnp:23:38-48: error: 'GenericPersistent' has no member named 'persistent'
  sandstorm/grain.capnp:44:31-41: error: Not defined: persistent
  sandstorm/grain.capnp:23:7-17: error: Type names must begin with a capital letter.
  sandstorm/supervisor.capnp:59:37-47: error: Not defined: Capability
capnpc: sandstorm/ip.capnp
  sandstorm/grain.capnp:23:38-48: error: 'GenericPersistent' has no member named 'persistent'
  sandstorm/grain.capnp:44:31-41: error: Not defined: persistent
  sandstorm/grain.capnp:23:7-17: error: Type names must begin with a capital letter.
capnpc: sandstorm/web-session.capnp
  sandstorm/grain.capnp:23:7-17: error: Type names must begin with a capital letter.
compile: capnp/rpc-test.c++
  /ekam-provider/canonical/capnp/rpc-test.c++:451:19: warning: 'makeRpcServer' is deprecated: Please transition to using a bootstrap interface instead. [-
    Wdeprecated-declarations]
          rpcServer(makeRpcServer(serverNetwork, restorer)) {}
                    ^
  /ekam-provider/canonical/capnp/rpc.h:391:18: note: 'makeRpcServer<capnproto_test::capnp::test::TestSturdyRefHostId, capnproto_test::capnp::test::TestSturdy
    RefObjectId, capnproto_test::capnp::test::TestProvisionId, capnproto_test::capnp::test::TestRecipientId, capnproto_test::capnp::test::TestThirdPartyCapId
    , capnproto_test::capnp::test::TestJoinResult>' has been explicitly marked deprecated here
  RpcSystem<VatId> makeRpcServer(
                   ^
  /ekam-provider/canonical/capnp/rpc-test.c++:475:22: warning: 'restore' is deprecated: Please transition to using a bootstrap interface instead. [-
    Wdeprecated-declarations]
      return rpcClient.restore(hostId, ref.getObjectId());
                       ^
  /ekam-provider/canonical/capnp/rpc.h:75:22: note: 'restore' has been explicitly marked deprecated here
    Capability::Client restore(typename VatId::Reader hostId, AnyPointer::Reader objectId)
```

--- a/Vagrantfile
+++ b/Vagrantfile
@@ -5,11 +5,8 @@
 VAGRANTFILE_API_VERSION = "2"
 
 Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
-  # We base ourselves off the trusty (Ubuntu 14.04) base box.
-  config.vm.box = "trusty64"
-
-  # The url from which to fetch that base box.
-  config.vm.box_url = "https://cloud-images.ubuntu.com/vagrant/trusty/current/trusty-server-cloudimg-amd64-vagrant-disk1.box"
+  # We base ourselves off a Debian fork of the official Debian jessie64 base box.
+  config.vm.box = "sandstorm/debian-jessie64"
 
   # We forward port 6080, the Sandstorm web port, so that developers can
   # visit their sandstorm app from their browser as local.sandstorm.io:6080
@@ -22,14 +19,10 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
   # work.
   config.vm.network :private_network, ip: "169.254.254.2"
 
-  # Use a shell script to "provision" the box. This install Sandstorm using
+  # Use a shell script to "provision" the box. This installs Sandstorm using
   # the bundled installer.
   config.vm.provision "shell",
-    inline: "cd /vagrant && echo localhost > /etc/hostname && hostname localhost && sudo ./install.sh -d -e && sudo sed --in-place='' --expression='s/^BIND_IP=.*/BIND_IP=0.0.0.0/' /opt/sandstorm/sandstorm.conf && sudo service sandstorm restart"
-
-  # Make the vagrant user part of the sandstorm group so that commands like
-  # `spk dev` work.
-  config.vm.provision "shell", inline: "usermod -a -G 'sandstorm' 'vagrant'"
+    inline: "cd /vagrant && echo localhost > /etc/hostname && hostname localhost && sudo OVERRIDE_DEFAULT_SERVER_USER=vagrant ./install.sh -d -e > /dev/null && sudo sed --in-place='' --expression='s/^BIND_IP=.*/BIND_IP=0.0.0.0/' /opt/sandstorm/sandstorm.conf && sudo service sandstorm restart && printf '\nYour server is online. It has the dev accounts feature enabled, so anyone can log in.\n\nDetails and customization instructions are available here:\n- https://github.com/sandstorm-io/sandstorm/wiki/Using-the-Vagrantfile\n\nVisit it at:\n  http://local.sandstorm.io:6080/'"
 
   # Use NFS for the /vagrant shared directory, for performance and
   # compatibility.

